src/Layouts/Partials/Layer.vala

```diff
@@ -472,32 +507,50 @@ public class Akira.Layouts.Partials.Layer : Gtk.ListBoxRow {
                         return false;
                     });
                 }
-            } else {
-                artboard.container.selection_mode = Gtk.SelectionMode.SINGLE;
 
-                window.main_window.right_sidebar.layers_panel.foreach (child => {
-                    if (child is Akira.Layouts.Partials.Artboard) {
-                        Akira.Layouts.Partials.Artboard artboard = (Akira.Layouts.Partials.Artboard) child;
+                return true;
+            }
+
+            if (artboard != null) {
+                artboard.container.selection_mode = Gtk.SelectionMode.SINGLE;
+            }
 
-                        window.main_window.right_sidebar.layers_panel.unselect_row (artboard);
-                        artboard.container.unselect_all ();
+            window.main_window.right_sidebar.layers_panel.foreach (child => {
+                if (child is Akira.Layouts.Partials.Artboard) {
+                    Akira.Layouts.Partials.Artboard artboard = (Akira.Layouts.Partials.Artboard) child;
 
-                        unselect_groups (artboard.container);
-                    }
-                });
+                    window.main_window.right_sidebar.layers_panel.unselect_row (artboard);
+                    artboard.container.unselect_all ();
 
-                if (layer_group != null) {
-                    artboard.container.selection_mode = Gtk.SelectionMode.NONE;
-                    artboard.container.unselect_row (layer_group);
+                    unselect_groups (artboard.container);
                 }
+            });
+
+            if (layer_group != null) {
+                artboard.container.selection_mode = Gtk.SelectionMode.NONE;
+                artboard.container.unselect_row (layer_group);
             }
 
-            activate ();
+            //activate ();
+            //window.main_window.right_sidebar.layers_panel.selection_mode = Gtk.SelectionMode.NONE;
```

Comment:
Leftover comment?

Reply:
Well, kinda. It is something related to artboards selection or something, so I'm not totally sure what was the purpose of it. Nevertheless, when we'll tackle the artboard thing we'll account for this, now I simply remove it.